React/Base/RCTBundleURLProvider.m

```diff
@@ -19,9 +19,18 @@
 static NSString *const kRCTEnableMinificationKey = @"RCT_enableMinification";
 
 static NSString *const kDefaultPort = @"8081";
+static NSString *ipGuess;
 
 @implementation RCTBundleURLProvider
 
++ (void)initialize
+{
+  NSString *ipPath = [[NSBundle mainBundle] pathForResource:@"ip" ofType:@"txt"];
+  NSString *ip = [NSString stringWithContentsOfFile:ipPath encoding:NSUTF8StringEncoding error:nil];
+  ip = [ip stringByTrimmingCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@"\n"]];
```

Comment:
Use `[NSCharacterSet  newlineCharacterSet]`. I'd also assign it immediately to `ipGuess` instead of making that a separate assignment.